clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -694,19 +694,18 @@ private void ensureProperTransactionalState() {
         if (transactionManager.isFenced())
             throw new ProducerFencedException("The current producer has been fenced off by a another producer using the same transactional id.");
```

Comment:
This message needs to be adjusted if we're handling the idempotent case as well.